### PR TITLE
Svelte: Fix reactivity loops & crashes

### PR DESCRIPTION
**Pull request by pingu-codes**
This PR implements the changes discussed in #484 which fixes bugs that occur from Svelte 5's new reactivity system.

### CODE REVIEW

File: packages/svelte/src/containers/xy-container/xy-container.svelte [hunk 1]
```diff
@@ -46,26 +48,39 @@
   })
 
   setContext('component', () => ({
-    update: (c: XYComponentCore<Datum>) => { config.components = [...config.components, c] },
+    update: (c: XYComponentCore<Datum>) => {
+      if (config.components?.some((e) => e === c)) return
```
Comment (rokotyan) [inline packages/svelte/src/containers/xy-container/xy-container.svelte:52]: We can probably use the `includes()` function here instead

File: packages/svelte/src/containers/xy-container/xy-container.svelte [hunk 2]
```diff
@@ -38,6 +38,8 @@
   let ref: HTMLDivElement
 
   $: chart?.setData(data, true)
+  // eslint-disable-next-line no-self-assign
+  $: config.components = config.components
```
Comment (rokotyan) [inline packages/svelte/src/containers/xy-container/xy-container.svelte:42]: Do you know if there are there any drawbacks of doing this, e.g. extra re-renders when nothing changes or something like that?

**Review by rokotyan — COMMENTED** [review]
Thanks for the contribution! Let a few minor comments

File: packages/svelte/src/containers/xy-container/xy-container.svelte (hunk 2 above)
Comment (pingu-codes) [reply packages/svelte/src/containers/xy-container/xy-container.svelte]: I'm not sure it actually fixes what it was originally supposed to. My original issue was that the chart wasn't rerendering when child component data changes but this doesn't actually fix that. I'm not sure if it's intended that when a child components data is set the whole graph rerenders.

If it is it'll need a different fix and I can look into it lmk :)

File: packages/svelte/src/containers/xy-container/xy-container.svelte (hunk 1 above)
Comment (pingu-codes) [reply packages/svelte/src/containers/xy-container/xy-container.svelte]: fixed

**Comment by pingu-codes** [comment]
I have hereby read the F5 CLA and agree to its terms

**Comment by lee00678** [comment]
Just released [1.5.0-svelte.0](https://www.npmjs.com/package/@unovis/ts/v/1.5.0-svelte.0). I tested it with a simple Svelte 5 project, seems to be working for me.

**Comment by lee00678** [comment]
@pingu-codes This PR looks good. Do you mind cleaning up the commit history (part of my fault as well)? You can either squash it or overwrite the history and do a force push to the branch.

**Comment by pingu-codes** [comment]
Yep, just squashed the commits!

**Comment by pingu-codes** [comment]
Do you think I should remove the changes to the package-lock.json on website too as it seems to include a lot of changes not really related to the fix?

**Comment by 50rayn** [comment]
I would rebase onto the latest `main` branch. After that, discard changes from `package-lock.json` and rerun an `npm ci`. And recheck if everything is ok and works as expected. Why `npm ci`, because `npm i` adds new dependencies and updates dependencies on a project.

**Comment by lee00678** [comment]
> Do you think I should remove the changes to the package-lock.json on website too as it seems to include a lot of changes not really related to the fix?

Yea, I would rebase from `main`. We did a major website upgrade so there will be a lot of updates in the website `package-lock.json`

**Review by lee00678 — APPROVED** [review]

**Comment by pingu-codes** [comment]
Did you handle the changes with your push @lee00678 sorry I've been really busy the last few days. I'm pretty bad at github tbh (mainly use graphite at my work) so I don't want to break anything if you've made the changes you need 😅

**Review by sumitkumar25 — APPROVED** [review]

**Pull request merged**